dot/network/service.go

```diff
@@ -344,6 +394,13 @@ func (s *Service) Stop() error {
 		logger.Error("Failed to close host", "error", err)
 	}
 
+	defer func() {
+		if r := recover(); r != nil {
+			logger.Error("failed to close telemetry service")
+		}
+	}()
+	close(s.doneNetworkTelemetry)
+
```

Comment:
Note: I added this to catch panic that happens (during testing) because stop is called twice:  
in dot/core/test_helpers.go line 183:
```
	if net, ok := cfg.Network.(*network.Service); ok {
		net.SetTransactionHandler(s)
		_ = net.Stop()
	}
```

and in dot/core/test_helpers.go line 208:
```
	t.Cleanup(func() {
		err := srvc.Stop()
		require.NoError(t, err)
	})
```
So, the service is stopped (when created as test service) in line 183, then stop is called again during test cleanup in line 208.

Please advise if I should remove one of these and remove the recover check, or if I should keep the recover check in place?

Reply:
You can rename `doneNetworkTelemetry` to `closeCh` to make it more generic to signal the close of service.
Using the channel you can also check if the service is closed or not and return early if it's already closed. This will avoid panic on multiple close calls.
Also alternatively you can wrap the close code in `sync.Once` which will only be called once.

Reply:
Also, Channel names in Golang end  usually with `Ch`